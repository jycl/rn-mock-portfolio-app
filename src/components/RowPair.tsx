import React from 'react';
import { View, Text, StyleSheet, ViewStyle } from 'react-native';

type PropsType = {
  lowerColor?: string;
  upper: string;
  lower: string | number;
  containerStyle?: ViewStyle;
};

/**
 * Common component for row item of larger font sitting on top of
 * smaller font in a column.
 */
export default function RowPair({
  lowerColor,
  upper,
  lower,
  containerStyle,
}: PropsType) {
  return (
    <View style={[styles.container, containerStyle]}>
      <Text style={styles.upper}>{upper}</Text>
      <Text style={[styles.lower, { color: lowerColor }]}>{lower}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'flex-end',
  },
  upper: {
    color: 'white',
    fontSize: 18,
  },
  lower: {
    fontSize: 14,
  },
});
