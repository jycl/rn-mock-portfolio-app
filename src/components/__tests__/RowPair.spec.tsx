import React from 'react';
import { create } from 'react-test-renderer';
import RowPair from '../RowPair';

describe('RowPair component', () => {
  it('should match snapshot given defined props', () => {
    const renderItem = create(<RowPair upper="Boss" lower="Underling" />);
    expect(renderItem.toJSON()).toMatchSnapshot();
  });

  it('should match snapshot given defined style props', () => {
    const renderItem = create(
      <RowPair
        upper="Rainbow"
        lower="Pot of gold"
        containerStyle={{ borderRadius: 12, padding: 16 }}
        lowerColor="#FFFFFF"
      />,
    );
    expect(renderItem.toJSON()).toMatchSnapshot();
  });
});
