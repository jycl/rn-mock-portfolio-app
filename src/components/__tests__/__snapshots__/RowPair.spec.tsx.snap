// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowPair component should match snapshot given defined props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
      }
    }
  >
    Boss
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
        Object {
          "color": undefined,
        },
      ]
    }
  >
    Underling
  </Text>
</View>
`;

exports[`RowPair component should match snapshot given defined style props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "flexDirection": "column",
      },
      Object {
        "borderRadius": 12,
        "padding": 16,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
      }
    }
  >
    Rainbow
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Pot of gold
  </Text>
</View>
`;
